File: app/src/main/java/com/takeaway/data/db/StatusConverter.java
//package com.takeaway.data.db;
//
//import androidx.room.TypeConverter;
//
//public class StatusConverter {
//
//    @TypeConverter
//    public static Status toStatus(int status) {
//        switch (status) {
//            case 0:
//                return OPEN;
//            case 1:
//                return ORDER_AHEAD;
//            case 2:
//                return CLOSED;
//        }
//        throw new IllegalArgumentException("Could not recognize status");
//    }
//
//    @TypeConverter
//    public static int toInteger(Status status) {
//        return status.ordinal();
//    }
//}

File: app/src/main/java/com/takeaway/utils/StatusConverter.kt
package com.takeaway.utils

import androidx.room.TypeConverter

class StatusConverter {

    @TypeConverter
    fun toStatus(status: Int): Status {
        when (status) {
            0 -> return Status.OPEN
            1 -> return Status.ORDER_AHEAD
            2 -> return Status.CLOSED
        }
        throw IllegalArgumentException("Could not recognize status")
    }

    @TypeConverter
    fun toInteger(status: Status): Int {
        return status.ordinal
    }
}
